Main CI red: Build & publish signed AAB

## Main CI red

**Repo**: alexsiri7/un-reminder
**SHA**: `aa20e20202610d1ff8a906a1a26466a4141c7a30`
**Run**: https://github.com/alexsiri7/un-reminder/actions/runs/24976350193
**Failed jobs**: Build & publish signed AAB

Auto-filed by `pipeline-health-cron.sh`. Main CI is a pipeline bottleneck, so archon has been fired immediately on this issue rather than queued. This issue is tagged `archon:in-progress` so the regular pickup cron will not double-fire.

### Steps
1. `gh run view 24976350193 --repo alexsiri7/un-reminder --log-failed`
2. Identify root cause (dep bump? toolchain? flaky test?)
3. Fix on a branch, open PR, ensure CI goes green

## 🔍 Investigation: Main CI red: Build & publish signed AAB

**Type**: `BUG`

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `HIGH` | Main CI is blocked on every push to main; no automated release can land until fixed |
| Complexity | `LOW` | Single line change in one workflow file, no code logic involved |
| Confidence | `HIGH` | Error message is unambiguous and git history shows the exact regression commit |

---

### Problem Statement

The Play Store upload step in `release.yml` fails with:

```
Only releases with status draft may be created on draft app.
```

PR #161 (`aa20e20`) reverted `status: draft` → `status: completed`, but the Play Store app is still in draft state and the API rejects non-draft uploads against a draft app.

---

### Root Cause Analysis

WHY: "Publish to Play Store internal track" step fails
↓ BECAUSE: `r0adkll/upload-google-play@v1` receives `status: completed`
  Evidence: `.github/workflows/release.yml:81`

↓ BECAUSE: PR #161 reverted the draft workaround without verifying the app had exited draft state
  Evidence: `aa20e20 ci: revert Play Store status from draft to completed (#161)`

↓ **ROOT CAUSE**: Play Store app is still in draft state — the API only accepts `status: draft` releases
  Evidence: `##[error]Only releases with status draft may be created on draft app.`

**Original fix**: `cf0a86f` (PR #120) correctly set `status: draft` for this reason. PR #161 undid it prematurely.

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | `.github/workflows/release.yml:81` | `status: completed` → `status: draft` |

<details>
<summary>📋 Detailed Implementation Steps</summary>

**File**: `.github/workflows/release.yml`, line 81

Current:
```yaml
          status: completed
```

Change to:
```yaml
          status: draft
```

This restores the state that was working before PR #161. When the app eventually exits draft state in Play Console, revert to `status: completed` — but only after verifying it in Play Console first.

</details>

---

### Validation

Push the fix to main and verify the "Build & publish signed AAB" job passes. Confirm the release appears in Play Console internal track as a draft.

---

### Next Step

To implement: `/implement-issue 162`

---
*Investigated by Claude • 2026-04-27*

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -78,7 +78,7 @@ jobs:
           packageName: net.interstellarai.unreminder
           releaseFiles: app/build/outputs/bundle/release/*.aab
           track: internal
-          status: completed
+          status: draft
           releaseName: ${{ env.VERSION_NAME }}
 
       - name: Build signed universal APK (for sideload)

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -78,6 +78,7 @@ jobs:
           packageName: net.interstellarai.unreminder
           releaseFiles: app/build/outputs/bundle/release/*.aab
           track: internal
+          # App is in Play Store draft state until first public release; use 'completed' after first publish
           status: draft
           releaseName: ${{ env.VERSION_NAME }}
 

